### ISSUE TITLE
En Garde! (1654660)

### ISSUE THREAD

**Issue by DomiStyle** [ValveSoftware/Proton#7025]
# Compatibility Report
- Name of the game with compatibility issues: En Garde!
- Steam AppID of the game: 1654660

## System Information
- GPU: AMD 6900 XT
- Video driver version: Mesa 23.1.6
- Kernel version: 6.4.10-201.fsync.fc38.x86_64
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/DomiStyle/7d57373b18c40b311157f2ffb612c8cf
- Proton version: Proton 8.0-3 and Proton Experimental

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-1654660_crash.log](https://github.com/ValveSoftware/Proton/files/12378851/steam-1654660_crash.log)

## Symptoms 
The game (and Steam) crash at the start of chapter 3 when using the default DirectX 11 backend. DirectX 12 can be used as a workaround.

## Reproduction
1. Play the the game until you reach chapter 3
2. Wait a few seconds at the beginning of chapter 3 (after the cutscene plays)
3. Game crashes

**Comment by Blisto91** [comment]
Hi there and thank you for the report.
I have not been able to reproduce this so far either with my 7900xtx or my rdna2 iGPU. Do you know if it is tied to any special settings or such?

I assume this area shown below is the one that will crash in for you?
<details>
  <summary>Screenshot</summary>
  
![image](https://github.com/ValveSoftware/Proton/assets/47954800/dea0de4e-3ceb-47eb-8adb-19b0c16183c6)
</details>

**Comment by DomiStyle** [comment]
Yes, the area is correct. Settings are all Ultra except anti-aliasing on FXAA and Motion blur is off.

After some more testing I figured out that the game has a memory leak with DX11. Not sure what I did to trigger this but it now happens on DX11 every time I start the game, even on the main menu. Played for a full hour before that with no issues. The timing just aligned perfectly with loading into the third chapter.

Took a video of it here:

https://github.com/ValveSoftware/Proton/assets/975131/66358e26-5e39-4549-959c-29aefb490e10

The game in DX11 fills up the entire 32GB of RAM in just a few seconds, on DX12 it runs without issues.

**Comment by Blisto91** [comment]
@DomiStyle Hi there. I'm Back again.
I have a suspicion it's probably this driver regression that was introduced with 23.1.6 https://gitlab.freedesktop.org/mesa/mesa/-/issues/9599 which can cause huge memory leaks in some games.

**Comment by DomiStyle** [comment]
Yes, that sounds just like it.

I was able to resolve it by switching from Fedora Steam to the Flatpak version, which uses a different Mesa version. The issue mentioned above doesn't happen there.

Thanks for looking into it, I finished En Garde with no other issue.
